utility.sh

```diff
@@ -116,6 +116,26 @@ get_resource_version_number() {
   eval echo "$""$UP"_VERSIONNUMBER
 }
 
+get_resource_version_key() {
+  if [ "$1" == "" ] || [ "$2" == "" ]; then
+    echo "Usage: shipctl get_resource_version_key RESOURCE_NAME KEY"
+    exit 99
+  fi
+  UP=$(get_resource_name "$1")
+  RESOURCE_META=$(get_resource_meta $UP)
+  if [ ! -d $RESOURCE_META ]; then
+    echo "IN directory not present for resource: $1"
+    exit 99
+  fi
+  RESOURCE_VERSION_FILE=$RESOURCE_META/version.json
+  if [ ! -f $RESOURCE_VERSION_FILE ]; then
+    echo "version.json not present for resource: $1"
+    exit 99
+  fi
+  VERSION_KEY_CMD="cat $RESOURCE_VERSION_FILE | jq '.version.$2'"
```

Comment:
@trriplejay I am operating on the version object inside the version.json file. Please let me know if the `get_resource_version_key` command should operate on the entire version.json file.

Reply:
can this be used to get the versionName? ( i know there is a specific function for that already, just curious)

can you post an example output of the whole version.json? i think operating on the version object within version.json is correct, just want to see if there's any reason that wouldn't be enough.

Reply:
yes this can be used to get the versionName. I have added a version.json from one of my jobs. The version object is present in version.json .

Reply:
ahh.. ok.. well shouldn't the values be accessed in `version.propertyBag.<arg>` except for versionName, versionNumber, versionId? 

try putting a custom value in a version of something, and getting it out with this tool, it doesn't seem like it would work.

for example:
```yml
- name: myImage
  type: image
  pointer:
    sourceName: trriplejay/simpleserver
  version:
    versionName: master.58
    env: production
    sha: abcdefg
```

Then, use that image as an IN to a runSh, and try 
```sh
shipctl get_resource_version_key utility myImage env
```
is that going to return `production` ?